src/arch/ia32/interrupts/pic8259a.rs

```diff
@@ -0,0 +1,248 @@
+    pub const fn slave(addr: u16, slave_pin: u8, offset: u8) -> Self {
+        if slave_pin >= 8 {
+            panic!("8259a: pin must be between 0 and 7");
+        }
+
+        if offset % 8 != 0 {
+            panic!("8259a: offset must be a multiple of 8.");
+        }
+
+        Self {
+            com_port: Port::new(addr),
+            data_port: Port::new(addr + 1),
+            pin: Some(slave_pin),
+            offset: offset,
+            role: PicRole::SLAVE,
+        }
+    }
+
+    /// Sends the Initialisation Command Word 1.
+    ///
+    /// # Arguments
+    ///
+    /// * `has_icw4` - If true, ICW4 will be expected.
+    /// * `adi` - CALL Address Inverval (true = 4, false = 8).
+    /// * `ltim` - Level Triggered mode (true) or Edge Triggered mode (false).
+    pub unsafe fn send_icw1(&self, has_icw4: bool, adi: bool, ltim: bool) {
+        let com = ICW1::new(has_icw4, self.role == PicRole::SINGLE, adi, ltim);
+        self.com_port.write(com.into());
+    }
+
+    /// Sends the Initialisation Command Word 2.
+    pub unsafe fn send_icw2(&self) {
+        self.data_port.write(self.offset);
+    }
+
+    /// Sends the Initialisation Command Word 3.
+    pub unsafe fn send_icw3(&self) {
+        match self.role {
+            PicRole::SINGLE => panic!("ICW3: Nothing to be sent in single mode."),
+            PicRole::MASTER => self
+                .data_port
+                .write(0_u8.set_bit(self.pin.unwrap().into(), true)),
+            PicRole::SLAVE => self.data_port.write(self.pin.unwrap()),
+        };
+    }
+
+    /// Sends the Initialisation Command Word 4.
+    ///
+    /// # Arguments
+    ///
+    /// * `aeoi` - Auto EOI mode.
+    /// * `buffered` - Buffered mode.
+    /// * `fully_nested` - Special fully nested mode.
+    pub unsafe fn send_icw4(&self, aeoi: bool, buffered: bool, fully_nested: bool) {
+        let com = 0_u8
+            .set_bit(0, true)
+            .set_bit(1, aeoi)
+            .set_bit(2, self.role == PicRole::MASTER)
+            .set_bit(3, buffered)
+            .set_bit(4, fully_nested);
+        self.data_port.write(com);
+    }
+
+    /// Send Operational Command Word 1.
+    ///
+    /// # Arguments
+    ///
+    /// * `mask` - IRQs bitmask (1 = IRQ disabled).
+    pub unsafe fn send_ocw1(&self, mask: u8) {
+        self.data_port.write(mask);
+    }
+
+    /// Acknowledge End Of Interrupt on PIC.
+    pub unsafe fn ack_eoi(&self) {
+        self.com_port.write(0x20_u8);
+    }
+}
+
+static MASTER_PIC: PIC8259a = PIC8259a::master(MASTER_PIC_COM, 2, PIC_OFFSET as u8);
+static SLAVE_PIC: PIC8259a = PIC8259a::slave(SLAVE_PIC_COM, 2, (PIC_OFFSET + 8) as u8);
+
+/// Setup both Master and Slave PICs. Sends Initialisation Commands and IRQs masks.
+///
+/// # Arguments
+///
+/// * `master_mask` - Master PIC IRQs bitmask (1 = IRQ disabled).
+/// * `slave_mask` - Slave PIC IRQs bitmask (1 = IRQ disabled).
+pub fn setup_pic(master_mask: u8, slave_mask: u8) {
```

Comment:
We should rename this function to "setup" don't we ? I mean, the `pic` name is already in the module name, since rust doesn't have namespace problems we could do that.

Reply:
Fixed in 1b7fa082f519efe83b62b109f000ef7bec0b1cda